backend/services/theme_service.py

```
@@ -23,9 +23,141 @@
+                
+                # Download the generated image
+                img_url = image_urls[0]
+                img_response = await client.get(img_url)
+                img_response.raise_for_status()
+                
+                logger.info("✅ RunWare IP-Adapter FaceID generation completed successfully")
+                return img_response.content
+                
+        except httpx.HTTPStatusError as e:
+            logger.error(f"❌ RunWare API error: {e.response.status_code} - {e.response.text}")
+            raise HTTPException(status_code=502, detail=f"RunWare API error: {e.response.status_code}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)